devtools/conda-envs/test-env.yaml

```diff
@@ -21,22 +21,16 @@ dependencies:
   - click
   - click-option-group
   - rdkit
-  - openff-utilities
   - openff-toolkit-base >=0.11.3
+  - openff-interchange-base
   - openff-forcefields
-  - openff-units
```

Comment:
Could you explain this change? I don't think we use Interchange directly in BespokeFit, but we do use Utilities and Units. I know Interchange will bring both in, but so does the Toolkit (which also brings in Interchange afaik). 

My preference is to use the environments to document the packages we actually use, rather than a collection of packages that will install the packages we use; I feel like this makes it easier to find the code we rely on without having to solve an environment, and it protects us against packaging changes if eg. one day Interchange and the Toolkit stop depending on Utilities.

Not a big deal in this case since we control everything, but seems like a good practice? Would be curious to hear your thoughts.

Reply:
Good question - I brought it Interchange because at some point in the process I was getting a `ModuleNotFoundError`, probably because `openff-toolkit-base` [doesn't bring](https://github.com/conda-forge/openff-toolkit-feedstock/blob/cc7ec1a92a431d5b5cebdf5e013ef2e0f22c0798/recipe/meta.yaml#L17-L40) in Interchange.

If you feel strongly about keeping the `openff-units` and `openff-utilities` dependencies explicitly in here, I'm fine taking them down. I see many of these as redundant lines given how tightly coupled these specific packages are - i.e. is it unsafe to assume that `openff-toolkit` will bring in `openff-forcefields` and the rest of our core utilities?

I see where you're coming from in general and I think it makes sense to explicitly declare the sharper and less stable dependencies, but for packages that are calcified into our ecosystem (i.e. `numpy`) I don't see much value in listing it. I also generally don't like digging through these files when they're very large.

> My preference is to use the environments to document the packages we actually use

Again I think this makes sense but it's not something we really follow - OpenMM is listed as a dependency but it's not imported, does that mean it's not actually required, or maybe it was imported at some point in the past and is no longer? I'm liable to not update this file when a dependency is removed (implicitly or explicitly).

Reply:
Yep, I see in #222 that Interchange is missing from the environment. Thanks for that!

Good points, I hadn't considered the cost of a very long environment file. I think we might rely on OpenMM via the Toolkit, even though BespokeFit has no direct imports of it. So it's clear that these lines are murky.